tests/virt_control_plane_test.go

```diff
@@ -56,228 +53,282 @@ var _ = Describe("KubeVirt control plane resilience", func() {
+				if _, ok := pod.Spec.NodeSelector[labelKey]; ok {
+					return false, fmt.Errorf("pod %s has still node selector %s", pod.Name, labelKey)
+				}
 			}
+			return true, nil
 		}
-		return true, nil
-	}
 
-	AfterEach(func() {
-		eventuallyWithTimeout(removeNodeSelectorFromDeployments)
-		eventuallyWithTimeout(cleanUpSelectedNode)
+		AfterEach(func() {
+			eventuallyWithTimeout(removeNodeSelectorFromDeployments)
+			eventuallyWithTimeout(cleanUpSelectedNode)
 
-		time.Sleep(WaitSecondsBeforeDeploymentCheck)
+			time.Sleep(WaitSecondsBeforeDeploymentCheck)
 
-		eventuallyWithTimeout(checkControlPlanePodsDontHaveNodeSelector)
-		eventuallyWithTimeout(waitForDeploymentsToStabilize)
-	})
+			eventuallyWithTimeout(checkControlPlanePodsDontHaveNodeSelector)
+			eventuallyWithTimeout(waitForDeploymentsToStabilize)
+		})
 
-	When("evicting pods of control plane, last eviction should fail", func() {
+		When("evicting pods of control plane", func() {
 
-		test := func(podName string) {
-			By(fmt.Sprintf("Try to evict all pods %s from node %s\n", podName, selectedNodeName))
-			podList, err := getPodList()
-			Expect(err).ToNot(HaveOccurred())
-			runningPods := getRunningReadyPods(podList, []string{podName})
-			for index, pod := range runningPods {
-				err = virtCli.CoreV1().Pods(tests.KubeVirtInstallNamespace).Evict(&v1beta1.Eviction{ObjectMeta: metav1.ObjectMeta{Name: pod.Name}})
-				if index < len(runningPods)-1 {
-					Expect(err).ToNot(HaveOccurred())
+			test := func(podName string) {
+				By(fmt.Sprintf("Try to evict all pods %s from node %s\n", podName, selectedNodeName))
+				podList, err := getPodList()
+				Expect(err).ToNot(HaveOccurred())
+				runningPods := getRunningReadyPods(podList, []string{podName})
+				for index, pod := range runningPods {
+					err = virtCli.CoreV1().Pods(tests.KubeVirtInstallNamespace).Evict(&v1beta1.Eviction{ObjectMeta: metav1.ObjectMeta{Name: pod.Name}})
+					if index < len(runningPods)-1 {
+						Expect(err).ToNot(HaveOccurred())
+					}
 				}
+				Expect(err).To(HaveOccurred(), "no error occurred on evict of last pod")
 			}
-			Expect(err).To(HaveOccurred(), "no error occurred on evict of last pod")
-		}
 
-		It("for virt-controller pods", func() { test("virt-controller") })
-		It("for virt-api pods", func() { test("virt-api") })
+			It("[test_id:2830]last eviction should fail for virt-controller pods", func() { test("virt-controller") })
+			It("[test_id:2799]last eviction should fail for virt-api pods", func() { test("virt-api") })
+
+		})
+
+	})
+
+	Context("control plane components check", func() {
+
+		When("control plane pods are running", func() {
+
+			It("[test_id:2806]virt-controller and virt-api pods have a pod disruption budget", func() {
+
+				By("check deployments")
+				deployments, err := deploymentsClient.List(metav1.ListOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				expectedDeployments := []string{"cdi-http-import-server", "virt-operator", "virt-api", "virt-controller"}
```

Comment:
it does not hurt, but why are we testing for cdi-http-import-server here? :thinking:

Reply:
Because that was described in the Polarion test case.

Reply:
ok. Don't want to discuss this further then :wink:

Reply:
:-D

Reply:
I think Zhe added it to polarion accidentally.. IMO, you don't need to check "cdi" and "virt-operator", because they are not related to this functional test :)
Please, remove it and I will update testcase in Polarion

Reply:
Done. /cc @slintes